### ISSUE TITLE
Calling the Invalidate of SKXamlCanvas do not raise PaintSurface event

### ISSUE THREAD

**Issue by lindexi** [unoplatform/uno#17326]
### Your feedback

Calling the Invalidate of SKXamlCanvas do not raise PaintSurface event. Nothing happened.

Why? Because the width and height of SKXamlCanvas is 0.

I believe that providing more info in such situations is more developer-friendly. 

The repro step:

1. The xaml code:

```xml
  <StackPanel
        HorizontalAlignment="Center"
        VerticalAlignment="Center">
    <windows:SKXamlCanvas x:Name="SkXamlCanvas" PaintSurface="SkXamlCanvas_OnPaintSurface"/>
  </StackPanel>
```

You can find the space of SKXamlCanvas will be compressed by StackPanel. It means that the ActualWidth and ActualHeight of SKXamlCanvas is 0.

2. The C# code:

```csharp
    public MainPage()
    {
        this.InitializeComponent();

        Loaded += MainPage_Loaded;
    }

    private void MainPage_Loaded(object sender, RoutedEventArgs e)
    {
        SkXamlCanvas.Invalidate();
    }

    private void SkXamlCanvas_OnPaintSurface(object? sender, SKPaintSurfaceEventArgs e)
    {
        System.Diagnostics.Debug.WriteLine("Raise PaintSurface");
    }
```

3. Run the project. And you will find it never output `Raise PaintSurface` to debug window

You can find my code in : https://github.com/lindexi/lindexi_gd/tree/e6093d810c34490fa522eaf1b71f0b959988fb8f/UnoDemo/LerkufufelchelDearrarfuguji

**Comment by jeromelaban** [comment]
This would be an issue to raise on the SkiaSharp repository.

**Comment by Youssef1313** [comment]
OnPaintSurface is only called when ActualWidth and ActualHeight are both > 0.

If you test on WinUI, does the SKXamlCanvas get positive values for ActualWidth and ActualHeight while they are zero on Uno?

In that case, it will be on our side and that will be https://github.com/unoplatform/uno/issues/16120

**Comment by lindexi** [comment]
@Youssef1313 Yeah, if I test on WinUI, the SKXamlCanvas get zero values for ActualWidth and ActualHeight.

**Comment by Youssef1313** [comment]
That should mean you won't be getting PaintSurface on WinUI. This is a behavior that SkiaSharp is responsible for. I'm not sure though why would you want to get the event for zero size. There is nothing to draw/render.

**Comment by lindexi** [comment]
@Youssef1313 If it can give more info to developer, I think it can help developer to debug why PaintSurface do not be raised.

**Comment by lindexi** [comment]
@Youssef1313 Do you think it is a good idea for me to write a debug or info level log message when developer call the `Invalidate` but the size is zero on SkiaSharp?

**Comment by Youssef1313** [comment]
Maybe. That's out of our control though and should be reported to SkiaSharp

**Comment by lindexi** [comment]
@Youssef1313 Thank you. Can we use UNO's ILogger in SKXamlCanvas.cs ?

**Comment by MartinZikmund** [comment]
The recommendation is to [utilize `SKCanvasElement` instead](https://platform.uno/docs/articles/controls/SKCanvasElement.html)
